orderBook.js

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable max-len */
+function reconcileOrder(existingBook, incomingOrder) {
+  let updatedBook = []
+
+  if (existingBook.length <= 0) {
+    return existingBook.concat(incomingOrder)
+  }
+
+  for (let i = 0; i < existingBook.length; i++) {
+    if (incomingOrder.type === existingBook[i].type) {
+      existingBook.push(incomingOrder)
+
+      return existingBook
+    }
+
+    if ((incomingOrder.type !== existingBook[i].type) && (incomingOrder.price !== existingBook[i].price)) {
+      existingBook.push(incomingOrder)
+
+      return existingBook
+    }
+    if ((incomingOrder.type !== existingBook[i].type) && (incomingOrder.price === existingBook[i].price) && (incomingOrder.quantity === existingBook[i].quantity)) {
```

Comment:
use else if statements since each one has its own return statement only one of them is expected to run.